### ISSUE TITLE
Is there any way to stop the wp-container-{id} class from being output in the Group block?

### ISSUE THREAD

**Issue by ddryo** [WordPress/gutenberg#38127]
For classic themes that are not FSE themes, I don't think `wp-container-{id}` is necessary.

I would like to have the ability to turn off this class, as I feel this makes the HTML look messy.

On the other hand, as I wrote in #38126, I need a class that can determine that it is a "Row" block.

**Comment by linseybrander** [comment]
I found it goes away if you remove the `wp_render_layout_support_flag` filter, like so in functions.php:

`remove_filter( 'render_block', 'wp_render_layout_support_flag', 10, 2 );`

This will also remove the inline css for align en flex in the footer:

`<style>.wp-container-61ed66e1ec861 .alignleft { float: left; margin-right: 2em; }.wp-container-61ed66e1ec861 .alignright { float: right; margin-left: 2em; }</style>`
`<style>.wp-container-61ed68714f919 {display: flex;gap: 0.5em;flex-wrap: wrap;align-items: center;flex-direction: column;align-items: flex-end;}.wp-container-61ed68714f919 > * { margin: 0; }`</style>

But unfortunately Buttons and Social Icons no longer add classnames like `is-content-justification-right` (why oh why), so you can't just add those align and flex styles yourself...

**Comment by MatzeKitt** [comment]
This should do the trick:

```php
remove_filter( 'render_block', 'gutenberg_render_layout_support_flag' );

add_filter( 'render_block', function( $block_content, $block ) {
	if ( $block['blockName'] === 'core/group' ) {
		return $block_content;
	}
	
	return gutenberg_render_layout_support_flag( $block_content, $block );
}, 10, 2 );
```

This will remove the function that creates this class in the first place and then adds it again for every other block type.

**Comment by ddryo** [comment]
@MatzeKitt 

Thank you. But that didn't work out.
I got an `Call to undefined functiongutenberg_render_layout_support_flag()` error.

**Comment by ddryo** [comment]
However, I was able to remove `wp-container-{id}` from all blocks with the following code.

```
remove_filter( 'render_block', 'wp_render_layout_support_flag', 10, 2 );
remove_filter( 'render_block', 'gutenberg_render_layout_support_flag', 10, 2 );
```

**Comment by nlstm-tm** [comment]
It took some digging but this is the proper way to do it specifically in the Gutenberg Editor

`wp.hooks.removeFilter('editor.BlockListBlock', 'core/editor/layout/with-layout-styles')`

Do this in your plain JS file enqueued in the Gutenberg Editor

**Comment by ddryo** [comment]
As it is used in various blocks, not just groups, I have found this specification necessary.

So I will close.

**Comment by jerome-toole** [comment]
In WordPress 6.1, a theme support option was added to opt out of these inlined layout styles and dynamic classes.

```
// Disable layout styles.

add_theme_support( 'disable-layout-styles' );
```

https://make.wordpress.org/core/2022/10/10/updated-editor-layout-support-in-6-1-after-refactor/

I found that the previous answer's solution (removing the `wp_render_layout_support_flag` filter) _mostly_ worked but also removed some classes from the markup that I did want such as alignment classes (e.g. `.is-content-justification-center`).
